include/pagmo/algorithms/pso_gen.hpp

```diff
@@ -0,0 +1,881 @@
+/* Copyright 2017 PaGMO development team
+
+This file is part of the PaGMO library.
+
+The PaGMO library is free software; you can redistribute it and/or modify
+it under the terms of either:
+
+  * the GNU Lesser General Public License as published by the Free
+    Software Foundation; either version 3 of the License, or (at your
+    option) any later version.
+
+or
+
+  * the GNU General Public License as published by the Free Software
+    Foundation; either version 3 of the License, or (at your option) any
+    later version.
+
+or both in parallel, as here.
+
+The PaGMO library is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received copies of the GNU General Public License and the
+GNU Lesser General Public License along with the PaGMO library.  If not,
+see https://www.gnu.org/licenses/. */
+
+#ifndef PAGMO_ALGORITHMS_pso_gen_HPP
+#define PAGMO_ALGORITHMS_pso_gen_HPP
+
+#include <iomanip>
+#include <random>
+#include <string>
+#include <tuple>
```

Comment:
I noticed that ``std::abs()`` is being used in the algorithm's code. Due to various C++ bullshit, when you use ``std::abs()`` you should include the following headers in order to be sure that the function is found: ``cstdlib``, ``cmath`` and ``cinttypes``.

More details here: http://en.cppreference.com/w/cpp/numeric/math/abs